File: TaskManager/TaskManager/Command/CreateIssueCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TaskManager;

namespace TaskManager
{
    public class CreateIssueCommand : ICommand
    {
        private IssueRepository _repo;
        private Issue _Issue;

        public CreateIssueCommand(IssueRepository repo, Issue issue)
        {
            _repo = repo;
            _Issue = issue;
        }

        public void Execute()
        {
            _repo.AddIssue(_Issue);
        }

        public void Undo()
        {
            _repo.DeleteIssue(_Issue.Id);
        }
    }
}

File: TaskManager/labFinal_164/command/CreateIssueCommand.java
package labFinal_164.command;

import labFinal_164.model.Issue;
import labFinal_164.repository.IssueRepository;

public class CreateIssueCommand implements ICommand {
    private final IssueRepository repository;
    private final Issue issue;
    private final String description;

    public CreateIssueCommand(IssueRepository repository, Issue issue) {
        this.repository = repository;
        this.issue = issue;
        this.description = "Create issue: " + issue.getId(); // Initialize AFTER issue is set
    }


    @Override
    public void execute() { repository.addIssue(issue); }
    @Override
    public void undo() { repository.deleteIssue(issue.getId()); }
    @Override
    public String getType() { return "CREATE"; }
    @Override
    public String getDescription() { return description; }




}
